subsys/bootloader/include/fw_metadata.h

```diff
@@ -86,6 +121,29 @@ static inline bool memeq_32(const void *expected, const void *actual,
 	return true;
 }
 
+static inline bool memeq_8(const void *expected, const void *actual, u32_t len)
+{
+	const u8_t *expected_8 = (const u8_t *) expected;
+	const u8_t *actual_8   = (const u8_t *) actual;
+
+	for (u32_t i = 0; i < len; i++) {
+		if (expected_8[i] != actual_8[i]) {
+			return false;
+		}
+	}
+	return true;
+}
+
+static inline bool memeq(const void *expected, const void *actual, u32_t len)
+{
+	if (((u32_t)expected & 3) || ((u32_t)actual & 3) || ((u32_t)len & 3)) {
+		/* Parameters are not word aligned. */
+		return memeq_8(expected, actual, len);
```

Comment:
Wouldn't it be more efficient to only to `memeq_8` on the non-aligned members, and then finish the rest of the aligned comparisons with the `memeq_32`?

Reply:
Yes, but it would be quite a bit more code, so I felt this was a good compromise. To give you an idea, imagine `expected` and `actual` starting _and_ ending on non-aligned addresses, and also being non-aligned by different amounts.

Reply:
Makes sense.